modules/core/src/main/java/org/apache/synapse/aspects/newstatistics/RuntimeStatisticCollector.java

```diff
@@ -0,0 +1,395 @@
+	 * @param endTime end time of the message flow
+	 */
+	public static void closeStatisticEntryForcefully(MessageContext msgCtx, long endTime) {
+		if (isStatisticsEnable) {
+			String statisticsTraceId = getStatisticsTraceId(msgCtx);
+			if (statisticsTraceId != null) {
+				if (runningStatistics.containsKey(statisticsTraceId)) {
+					StatisticsEntry entry = runningStatistics.get(statisticsTraceId);
+					endMessageFlow(msgCtx, statisticsTraceId, entry, endTime, true);
+				}
+			}
+		}
+	}
+
+	/**
+	 * Ends statistics collection for the message flow. If entry is successfully completed ending
+	 * its statistics collection statistics store is updated with new statistics data. Then entry
+	 * is removed from the running statistic map
+	 *
+	 * @param messageContext   message context
+	 * @param statisticTraceId statistic trace id for the message log
+	 * @param statisticsEntry  statistic entry to be closed
+	 * @param endTime          end time of the message flow
+	 */
+	private synchronized static void endMessageFlow(MessageContext messageContext, String statisticTraceId,
+	                                                StatisticsEntry statisticsEntry, long endTime,
+	                                                boolean closeForceFully) {
+		boolean isMessageFlowEnded = statisticsEntry.endAll(endTime, closeForceFully);
+		if (isMessageFlowEnded) {
+			if (log.isDebugEnabled()) {
+				log.debug("Statistic collection is ended for the message flow with statistic " +
+				          "trace Id :" + statisticTraceId);
+			}
+			//statisticsStore.update(statisticsEntry.getMessageFlowLogs());
+			messageContext.getEnvironment().getCompletedStatisticStore()
+			              .putCompletedStatisticEntry(statisticsEntry.getMessageFlowLogs());
+			runningStatistics.remove(statisticTraceId);
+		}
+	}
+
+	/**
+	 * Set statistics trace Id for statistic collection
+	 *
+	 * @param msgCtx message context
+	 */
+	public static void setStatisticsTraceId(MessageContext msgCtx) {
+		if (msgCtx.getProperty(SynapseConstants.NEW_STATISTICS_ID) == null) {
+			msgCtx.setProperty(SynapseConstants.NEW_STATISTICS_ID, msgCtx.getMessageID());
+		}
+	}
+
+	/**
+	 * Returns statistics trace id corresponding to the message context
+	 *
+	 * @param msgCtx message context
+	 * @return statistics trace id
+	 */
+	private static String getStatisticsTraceId(MessageContext msgCtx) {
+		return (String) msgCtx.getProperty(SynapseConstants.NEW_STATISTICS_ID);
+	}
+
+	/**
+	 * Returns cloned message identification number for the specified  message context. If message
+	 * context is not a cloned one default value of -1 is sent
+	 *
+	 * @param msgCtx message context
+	 * @return cloned message identification number
+	 */
+	private static int getMsgId(MessageContext msgCtx) {
+		if (msgCtx.getProperty(SynapseConstants.NEW_STATISTICS_MESSAGE_ID) != null) {
+			return (Integer) msgCtx.getProperty(SynapseConstants.NEW_STATISTICS_MESSAGE_ID);
+		} else {
+			return -1;
```

Comment:
Would have been ideal if a constant was used upon the values like "-1", which would then help you improve the maintainability of the code.